heron/ckptmgr/src/java/com/twitter/heron/ckptmgr/backend/IBackend.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.ckptmgr.backend;
+
+import java.util.Map;
+
+import com.twitter.heron.proto.ckptmgr.CheckpointManager;
+
+public interface IBackend {
```

Comment:
I've done most of the changes that you requested. Working on unittests. Meanwhile can you please take a look again?